content/reference/cd-supporting-new-ci.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Supporting A New CI System
+expanded_url: /reference/cd/
+menu:
+  reference:
+    parent: cd
+---
+
+## Overview
+
+The Pulumi CLI already supports enhanced metadata for several popular CI/CD systems. We have tried to make it easy for contributors to add/update support for CI systems. This document walks-through how you can modify the Pulumi CLI to support your own CI system or perhaps even update an existing one.
+
+The detection of metadata in a CI environment depends on some key environment variables that a CI system injects into the build environment of their build agents. The Pulumi CLI uses those environment variables to set try and determine the values for the following properties:
+
+- CI System Name
+- Build Type
+- Build URL
+- Commit SHA
+- PR Number
+- Commit Message
+
+The above metadata about your CI environment is then used to provide an enhanced experience in the Pulumi Console when you look at your stack activity log. The metadata from your CI environment combined with the information about your Git repository allows us to provide links to things such as the Pull Request on github.com or the commit that triggered a PR build.
+
+![A Stack update entry in the Pulumi Console](/images/reference/supporting-new-ci/stack-update.png)
+
+In order to add support for your CI system, you should be somewhat familiar in working with [Go](https://golang.org/).
+
+## How Does It Work?
+
+There are 2 parts to the CLI for it to be able to correctly detect the environment variables of a CI system.
+
+1. Detecting in which CI system the Pulumi CLI is currently running
+1. Detecting additional environment variables for a specific CI system
+
+In the `pulumi/pulumi` repo the `pkg/util/ciutil` contains all of the logic necessary for the above. Every stack `preview` or `update` calls the functions exported from this package.
```

Comment:
Add a link around the `pulumi/pulum` repo? And also add links for any files or directories mentioned throughout this article?

Reply:
Done.